SPEC.md

```diff
@@ -5286,6 +5295,97 @@ Example output:
 </p>
 </details>
 
+### Runtime Access to Requirements, Hints, and Metadata
+
+The `requirements` and `hints` sections comprise resource requests to the execution engine. But these requests can be [specified or overridden at runtime](#specifying--overriding-requirements-and-hints), and the execution engine has some latitude in whether and how it fulfills them. Thus, the workflow developer may wish to know exactly what resources are available at runtime, such as:
+
+* What are the actual resource allocations. For example, a task may request at least `8 GiB` of memory but may be able to use more memory if it is available.
+* The task metadata, to avoid duplication. For example, the task may wish to write log messages with the task's name and description without having to duplicate the information in the task's `meta` section.
+* The runtime engine may also choose to provide additional information at runtime.
+
+This information is provided by the `task` variable, which is implicitly defined by the execution engine. The type of `task` is a [scoped type](#scoped-types) with the following members:
+
+* `name`: The task name.
```

Comment:
Should we add `time` to the list? This is relevant for SLURM. I guess it also matters for the cloud backends. Can somebody who uses cloud backends comment on this? If it is only relevant for SLURM then there is no need to add it.

Reply:
As in the maximum runtime for a task? It could be useful to prevent tasks getting stuck and running much longer than expected but it might be something that isn't very portable. A task that should take no more than 15 minutes in one environment might legitimately take 40 minutes in a different environment. It might be something that would be more useful if added as a hint for that environment?

Reply:
Yes, you are right. A hint is much better.

Reply:
This PR is about what information is both useful and can be provided at runtime, regardless if it originates from a requirement, hint, or neither (I.e. an inherent property of the task or the runtime environment).

If time (start time? max allowed runtime? Both? What are the units/type/format?) is useful then we should include it.

Reply:
Max run time might be useful in Seconds. But that might be more of a hint.

Reply:
I added the following: `end_time`: An `Int?` whose value is the time by which the task must be completed, as a [Unix time stamp](https://en.wikipedia.org/wiki/Unix_time). A value of `0` means that the execution engine does not impose a time limit. A value of `None` means that the execution engine cannot determine whether the runtime of the task is limited. A positive value is a guarantee that the task will be preempted at the specified time, but is *not* a guarantee that the task won't be preempted earlier.

My reasoning is that "max time" is less precise because there could be a delay between when the clock starts and when the execution starts, whereas with a fixed point in time both the execution engine and the executing task will be on the same page as to when the task will be killed. The max runtime can be (approximately) back-calculated by subtracting the current time from `end_time` as the first step of the task.

Thoughts?